pybv/io.py

```diff
@@ -229,18 +229,13 @@ def _write_vmrk_file(vmrk_fname, eeg_fname, events, meas_date):
 
 def _scale_data_to_unit(data, unit):
     """Scale `data` in Volts to `data` in `unit`."""
-    if unit == 'V':
-        return data
-    elif unit == 'mV':
-        return data * 1e3
-    elif unit in ('µV', 'uV'):
-        return data * 1e6
-    elif unit == 'nV':
-        return data * 1e9
-    else:
-        raise ValueError(
-            f'Encountered unsupported unit: {unit}'
-            f'\nUse one of the following: {SUPPORTED_UNITS}')
+    scale = SUPPORTED_UNITS.get(unit, None)
+    if not isinstance(scale, float):
```

Comment:
Shouldn't `if scale is not None` also work?

Reply:
yes, it should work equivalently, but I found isinstance(..., float) more specific, because we do definitely need a float next :thinking:

Reply:
But you either get a `float` or `None`, so I find this instance check unnecessary. I'd avoid `isinstance` whenever possible.

Reply:
I am fine with changing this (will do), but why would you avoid isinstance?

Reply:
It isn't Pythonic IMO, because as a dynamically typed language I think code should rely on duck typing as far as possible. It is not always possible, so `isinstance` can't be avoided every time, but in this example a comparison with `None` is completely sufficient.

Reply:
thanks for the explanation, makes sense!